tests/testsuite.src/syn_file.at

```diff
@@ -2282,3 +2282,74 @@ AT_CHECK([$COMPILE_ONLY prog.cob], [1], [],
 prog.cob:12: error: syntax error, unexpected INPUT, expecting Identifier
 ])
 AT_CLEANUP
+
+# Verify that FD GLOBAL will create default handlers in sub-programs, but
+# only in its own sub-programs
+
+AT_SETUP([Scope of FD GLOBAL in nested programs])
+AT_KEYWORDS([FD GLOBAL nested])
+AT_DATA([prog.cob],[
+ IDENTIFICATION DIVISION.
+ PROGRAM-ID. NESTED-FILES.
+ PROCEDURE DIVISION.
+   CALL    "NESTED-PROGRAM-1".
+   CALL    "NESTED-PROGRAM-2".
+   STOP RUN.
+
+ IDENTIFICATION DIVISION.
+ PROGRAM-ID. NESTED-PROGRAM-1.
+ ENVIRONMENT DIVISION.
+ INPUT-OUTPUT SECTION.
+ FILE-CONTROL.
+   SELECT FILE-EXT ASSIGN TO EXTERNAL FILEEXT01.
+ DATA DIVISION.
+ FILE SECTION.
+ FD FILE-EXT GLOBAL
+           RECORD IS VARYING IN SIZE  CHARACTERS.
+ 01 FILE-EXT-REC PIC X(347).
+ PROCEDURE DIVISION.
+   CALL    "NESTED-PROGRAM-1-1".
+   CALL    "NESTED-PROGRAM-1-2".
+   GOBACK.
+
+ IDENTIFICATION DIVISION.
+ PROGRAM-ID. NESTED-PROGRAM-1-1.
+ DATA DIVISION.
+ WORKING-STORAGE SECTION.
+ 01 FILE-EXT-DAT PIC X(347).
+ PROCEDURE DIVISION.
+   NESTED-PROGRAM .
+   WRITE FILE-EXT-REC FROM FILE-EXT-DAT.
+   GOBACK.
+ END PROGRAM NESTED-PROGRAM-1-1.
+ 
+ IDENTIFICATION DIVISION.
+ PROGRAM-ID. NESTED-PROGRAM-1-2.
+ DATA DIVISION.
+ WORKING-STORAGE SECTION.
+ 01 FILE-EXT-DAT PIC X(347).
+ PROCEDURE DIVISION.
+   NESTED-PROGRAM .
+   WRITE FILE-EXT-REC FROM FILE-EXT-DAT.
+   GOBACK.
+ END PROGRAM NESTED-PROGRAM-1-2.
+ 
+ END PROGRAM NESTED-PROGRAM-1.
+
+ IDENTIFICATION DIVISION.
+ PROGRAM-ID. NESTED-PROGRAM-2.
+ DATA DIVISION.
+ WORKING-STORAGE SECTION.
+ PROCEDURE DIVISION.
+   GOBACK.
+ END PROGRAM NESTED-PROGRAM-2.
+ 
+ END PROGRAM NESTED-FILES.
+])
+AT_CHECK([$COBC -F -C prog.cob], [0], [], [])
+AT_CHECK([grep 'Paragraph Default Error Handler' prog.c], [0], [  /* Line: 0         : Paragraph Default Error Handler   : prog.cob */
+  /* Line: 0         : Paragraph Default Error Handler   : prog.cob */
+  /* Line: 0         : Paragraph Default Error Handler   : prog.cob */
+], [])
+AT_CLEANUP
```

Comment:
Please also add a `$COMPILE` - Somehow I think it is likely best to move that run_file.at (would still have failed the `$COMPILE`) and verify with an error in the most nested entry that the default handler is triggered.
And if we have that test we wouldn't need the syn test any more.

Reply:
...
and please "duplicate" it for a `DECLARATIVES` in the outer program, to verify not only the default handler but the file handler in any case can be executed.

Reply:
I added a `DECLARATIVES`, but I am not sure of what the expected behavior should be.

Right now, it looks like the handlers are only generated where they are declared, and not in subprograms. Should they be duplicated in all sub-program inheriting from the files ?

Also, the bug only appears on `WRITE` instructions, and I don't really know how to generate an error easily on a `WRITE` that would trigger the handler execution at runtime.